src/Analysis/Engine/Test/LanguageServerTests.cs

```diff
@@ -1185,6 +1198,22 @@ private class GetAllExtension : ILanguageServerExtension {
             refs.Select(r => $"{r._kind ?? ReferenceKind.Reference};{r.range}").Should().Contain(contains).And.NotContain(excludes);
         }
 
+        public static async Task AssertLineFormat(Server s, TextDocumentIdentifier document, int line, string ch, string newText, SourceLocation start, SourceLocation end) {
+            var edits = await s.DocumentOnTypeFormatting(new DocumentOnTypeFormattingParams {
```

Comment:
We have `ServerExtension` type that accumulates helper methods that help with simulating server activity. You can add `SendDocumentOnTypeFormatting(Uri, Position, char)` helper for it and call it directly from the test.

Reply:
I implemented this, if you'd like to take a look.